frontend/src/components/applied-filters-bar/applied-filters-bar.tsx

```diff
@@ -0,0 +1,221 @@
+import { Chip } from '@mui/material';
+import {
+  removeBrandDetails,
+  removeRangeValue,
+  resetAllFilters,
+  setBrandDetailsValue,
+  setCheckListValue,
+} from '@store/car-filter/slice';
+import { useLazyGetModelsOfBrandQuery } from '@store/queries/cars';
+import {
+  selectAppliedBrands,
+  selectNormalizedBrands,
+  selectNormalizedOptionsInAutocompleteType,
+} from '@store/selectors/car-filter-selectors';
+
+import styles from './styles.module.scss';
+
+const AppliedFiltersBar = (): ReactElement => {
+  const dispatch = useAppDispatch();
+
+  const { rangeFilters, checkLists, brandDetails } = useAppSelector(
+    (state) => state.carFilter,
+  );
+  const [appliedRanges, setAppliedRanges] = useState<string[][]>();
+  const [appliedCheckboxes, setAppliedCheckboxes] = useState<
+    AppliedFilterType[]
+  >([]);
+  const [appliedBrandDetails, setAppliedBrandDetails] =
+    useState<AppliedFilterType[]>();
+
+  const [normalizedModels, setNormalizedModels] = useState<{
+    [p: string]: ModelType;
+  }>({});
+
+  const normalizedOptions = useAppSelector(
+    selectNormalizedOptionsInAutocompleteType,
+  );
+
+  const normalizedBrands = useAppSelector(selectNormalizedBrands);
+
+  const allAppliedBrands = useAppSelector(selectAppliedBrands);
+
+  const [getModelsOfBrand, models] = useLazyGetModelsOfBrandQuery();
+
+  useEffect(() => {
+    allAppliedBrands.forEach((brandId) => {
+      getModelsOfBrand(brandId, true);
+    });
+  }, [brandDetails]);
+
+  useEffect(() => {
+    if (!models.data) return;
+
+    const newModels = models.data.reduce(
+      (obj, item) => Object.assign(obj, { [item.id]: item }),
+      {},
+    );
+    setNormalizedModels({ ...normalizedModels, ...newModels });
+  }, [models]);
+
+  const handleRangeDelete = (rangeName: string): void => {
+    dispatch(removeRangeValue(rangeName));
+  };
+
+  const handleCheckboxDelete = (filterName: string, id: string): void => {
+    const newList = checkLists[filterName as CheckListsNames].filter(
+      (item) => item !== id,
+    );
+
+    dispatch(setCheckListValue({ filterName: filterName, value: newList }));
+  };
+
+  const handleBrandDetailDelete = (filterName: string, id: string): void => {
+    if (filterName === 'brandId') {
+      const detail = brandDetails.filter((item) => item.brandId === id)[0];
+      dispatch(removeBrandDetails(detail.id));
+    }
+    if (filterName === 'modelId') {
+      const detail = brandDetails.filter((item) =>
+        item.modelIds.includes(id),
+      )[0];
+      const newDetail = {
+        ...detail,
+        modelIds: detail.modelIds.filter((modelId) => modelId !== id),
+      };
+      dispatch(setBrandDetailsValue(newDetail));
+    }
+  };
+
+  useEffect(() => {
+    const ranges = Object.entries(rangeFilters)
+      .map((item) => {
+        const label = Object.values(item[1])
+          .filter((value) => value !== '')
+          .sort((a, b) => +a - +b)
+          .join(' - ');
```

Comment:
Some JS syntax sugar :eye::tongue::eye:

```suggestion
      .map(([item]) => {
        const label = Object.values(item)
          .filter(Boolean)
          .sort((a, b) => +a - +b)
          .join(' - ');
```